src/NewRecipePage.tsx

```diff
@@ -0,0 +1,495 @@
+import React, { useEffect, useState } from "react";
+import "antd/dist/antd.min.css";
+import "./NewRecipe.css";
+import { Recipe } from "./recipe";
+import {
+  Button,
+  Rate,
+  Form,
+  Input,
+  Modal,
+  Checkbox,
+  Image,
+  Result,
+  notification,
+} from "antd";
+import { DeleteOutlined, PlusOutlined } from "@ant-design/icons";
+import { NotificationPlacement } from "antd/lib/notification";
+
+export default function NewRecipe() {
+  const { TextArea, Search } = Input;
+  const [visible, setVisible] = useState(false);
+  const [allRecipes, setAllRecipes] = useState<Recipe[]>([]);
+  const [someRecipes, setSomeRecipes] = useState<Recipe[]>([]);
+  const [loadedRecipes, setLoadedRecipes] = useState<Recipe[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [form] = Form.useForm();
+  const [modalForm] = Form.useForm();
```

Comment:
If you pass a type to useForm, then you won't have to deal with the forms being `any` types